sdk/appcenter-distribute/src/main/java/com/microsoft/appcenter/distribute/ReleaseDownloadListener.java

```diff
@@ -0,0 +1,179 @@
+ */
+class ReleaseDownloadListener implements ReleaseDownloader.Listener {
+
+    /**
+     * Context.
+     */
+    protected Context mContext;
+
+    /**
+     * Last download progress dialog that was shown.
+     * Android 8 deprecates this dialog but only reason is that they want us to use a non modal
+     * progress indicator while we actually use it to be a modal dialog for forced update.
+     * They will always keep this dialog to remain compatible but just mark it deprecated.
+     */
+    @SuppressWarnings({"deprecation", "RedundantSuppression"})
+    private android.app.ProgressDialog mProgressDialog;
+
+    /**
+     * Private field to store information about release we are currently working with.
+     */
+    private ReleaseDetails mReleaseDetails;
+
+    ReleaseDownloadListener(@NonNull Context context, @NonNull ReleaseDetails releaseDetails) {
+        mContext = context;
+        mReleaseDetails = releaseDetails;
+    }
+
+    @Override
+    public void onStart(long enqueueTime) {
+        SharedPreferencesManager.putInt(PREFERENCE_KEY_DOWNLOAD_STATE, DOWNLOAD_STATE_ENQUEUED);
+        SharedPreferencesManager.putLong(PREFERENCE_KEY_DOWNLOAD_TIME, enqueueTime);
+    }
+
+    @Override
+    public boolean onProgress(final long currentSize, final long totalSize) {
+        AppCenterLog.verbose(LOG_TAG, "downloadedBytes=" + currentSize + " totalBytes=" + totalSize);
+
+        /* If file size is known update downloadProgress bar. */
+        if (mProgressDialog != null && totalSize >= 0) {
+            HandlerUtils.runOnUiThread(new Runnable() {
+
+                @Override
+                public void run() {
+
+                    /* When we switch from indeterminate to determinate */
+                    if (mProgressDialog.isIndeterminate()) {
+
+                        /* Configure the progress dialog determinate style. */
+                        mProgressDialog.setProgressPercentFormat(NumberFormat.getPercentInstance());
+                        mProgressDialog.setProgressNumberFormat(mContext.getString(R.string.appcenter_distribute_download_progress_number_format));
+                        mProgressDialog.setIndeterminate(false);
+                        mProgressDialog.setMax((int) (totalSize / MEBIBYTE_IN_BYTES));
+                    }
+                    mProgressDialog.setProgress((int) (currentSize / MEBIBYTE_IN_BYTES));
+                }
+            });
+            return true;
+        }
+        return false;
+    }
+
+    @Override
+    public boolean onComplete(@NonNull Uri localUri) {
+        Intent intent = getInstallIntent(localUri);
+        if (intent.resolveActivity(mContext.getPackageManager()) == null) {
+            return false;
+        }
+
+        /* Check if app should install now. */
+        if (!Distribute.getInstance().notifyDownload(mReleaseDetails, intent)) {
+
+            /*
+             * This start call triggers strict mode in U.I. thread so it
```

Comment:
What thread are we now that we moved the code?

Reply:
It's background. Added annotation